### PR TITLE
Twinkles dc2 branch

### PR DESCRIPTION
**Pull request by jbkalmbach**
Code to add sprinkler to DC2 Run 1.1

### CODE REVIEW

File: python/desc/sims/GCRCatSimInterface/CatalogClasses.py
Comment (danielsf) [inline python/desc/sims/GCRCatSimInterface/CatalogClasses.py]: You can achieve this by adding `.astype(float)` to the end of line 163.  Not sure if that will be faster, but I suspect that it will.

File: python/desc/sims/GCRCatSimInterface/DatabaseEmulator.py
Comment (danielsf) [inline python/desc/sims/GCRCatSimInterface/DatabaseEmulator.py]: Why did you need to add these?

I *thought* I had implement the `CompoundDESCQAInstanceCatalog` class so that it evaluated the equality of `DESCQAObject`s on things that were relevant to `DESCQAObjects`.  Was something failing without these member variables?

File: python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py
Comment (danielsf) [inline python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py]: We will want this to be an argument passed in by the user so that Tom can redirect to wherever he puts that AGN parameters database for the final run.  `test_agn.db` is just a cartoon database I created for testing the AGN `DESCQAObject` during development.

File: python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py
Comment (danielsf) [inline python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py:151]: We have not thought about how the sprinkler will interact with the knots catalog.

The knots code is meant to add more interesting structure to galaxy disks.  There is currently a non-zero probability that it will try to add knots to a galaxy that has been replaced by the sprinkler.  In that case, the knots will be added based on wrong assumptions about the size and SED of the galaxy.  Probably, the safest thing to do at this point is add code to prevent the knots catalogs from touching the sprinkled galaxies.  Maybe we can add a column to all `DESCQAObject`s named `sprinkled`.  This will be an int that defaults to 0, but gets set to 1 by the sprinkler.  We can then modify the knots catalog to ignore any galaxy with `sprinkled==1`.

Thoughts?

File: python/desc/sims/GCRCatSimInterface/TwinklesClasses.py
Comment (danielsf) [inline python/desc/sims/GCRCatSimInterface/TwinklesClasses.py]: Again: I think we should replace this with `if` block that raise exceptions explicitly telling the user that the relevant file doesn't exist

File: python/desc/sims/GCRCatSimInterface/TwinklesClasses.py
Comment (danielsf) [inline python/desc/sims/GCRCatSimInterface/TwinklesClasses.py]: Remove commented code

File: python/desc/sims/GCRCatSimInterface/TwinklesClasses.py
Comment (danielsf) [inline python/desc/sims/GCRCatSimInterface/TwinklesClasses.py]: Go ahead and remove this block of comments, since this repo should never be used to get galaxies from fatboy

File: python/desc/sims/GCRCatSimInterface/TwinklesClasses.py
Comment (danielsf) [inline python/desc/sims/GCRCatSimInterface/TwinklesClasses.py]: I would remove the commented out `column_outputs` in this and the class below

File: python/desc/sims/GCRCatSimInterface/TwinklesClasses.py
Comment (danielsf) [inline python/desc/sims/GCRCatSimInterface/TwinklesClasses.py]: Add a comment here pointing out why these lines are special

**Review by danielsf — CHANGES_REQUESTED** [review]
This looks good to me.

The only reason I did not approve it is because I realized that we don't know what we are doing with respect to the knots objects that @EiffL added to the framework.  We should probably figure out how we are handling those before we mark this as 'Delivered'.

File: python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py
Comment (danielsf) [reply python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py:151]: @EiffL This is what I was referring to

File: python/desc/sims/GCRCatSimInterface/DatabaseEmulator.py
Comment (jbkalmbach) [reply python/desc/sims/GCRCatSimInterface/DatabaseEmulator.py]: I don't think I need those. I think I was trying to solve something in the compound Instance catalog and this was an attempted solution I didn't remove. I can't take those out now.

File: python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py
Comment (danielsf) [reply python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py:151]: Turns out, we won't be using knots in Run 1.1

@jbkalmbach: If you need someone to just approve this, you can ping me.

I will go back to minding my own business

Comment (EiffL) [reply python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py:151]: @danielsf  oh ok, I see, yes it makes a lot of sense to not add knots to the sprinkled galaxies. We can open an issue on this after the code gets merged if we don't want to delay the sprinkler at this point

**Review by danielsf — APPROVED** [review]
good to merge

**Pull request merged**
